lib/gooddata/lcm/actions/rename_existing_client_projects.rb

```diff
@@ -0,0 +1,79 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_action'
+
+module GoodData
+  module LCM2
+    class RenameExistingClientProjects < BaseAction
+      DESCRIPTION = 'Rename Existing Client Projects'
+
+      PARAMS = define_params(self) do
+        description 'Client Used for Connecting to GD'
+        param :gdc_gd_client, instance_of(Type::GdClientType), required: true
+
+        description 'Clients'
+        param :clients, array_of(instance_of(Type::HashType)), required: true, generated: true
+
+        description 'Organization Name'
+        param :organization, instance_of(Type::StringType), required: true
+      end
+
+      RESULT_HEADER = [
+        :id,
+        :pid,
+        :old_title,
+        :new_title
+      ]
+
+      class << self
+        def call(params)
+          client = params.gdc_gd_client
+
+          domain_name = params.organization || params.domain
+          domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
+
+          results = []
+          params.clients.each do |c|
+            begin
+              domain_client = domain.clients(c[:id])
+            rescue
+              next
+            end
+
+            begin
+              c_project = domain_client.project
```

Comment:
About fetching all client projects, I just want to make it clear before discussing. In this action, we will try to fetch projects of all clients which are mention in **input source**, not all clients in segment. If we use information from `PurgeClients`, we will need more memories to cache all project information of all clients in every segments while the clients need to edit project title in **input source** is quite small and it's a rare case (based on info of the reporter). So I think this implementation is acceptable. Agree?